src/extension/ncwms/src/main/java/au/org/emii/geoserver/wms/GeoserverUrlIndex.java

```diff
@@ -144,7 +144,11 @@ public Map<String, Map<String, Set<String>>> getUniqueDates(LayerDescriptor laye
     private Filter cqlFilterForTimestamp(String timestamp, String timeFieldName) {
         LOGGER.log(Level.INFO, String.format("Returning cql for timestamp '%s'", timestamp));
         try {
-            return CQL.toFilter(String.format("%s = '%s'", timeFieldName, timestamp));
+
+            DateTime time = new DateTime(timestamp);
+            DateTimeFormatter formatter = DateTimeFormat.forPattern("yyyy-MM-dd HH:mm:ss.SSS");  // Format required by geotools 29.x
```

Comment:
It is thread-safe no need to create this object on each call

Reply:
I made the formatter a class property